records_mover/db/postgres/postgres_copy_options.py

```diff
@@ -0,0 +1,336 @@
+from ...utils import quiet_remove
+from ...records.hints import cant_handle_hint
+from typing import Dict, Set
+from ...records.load_plan import RecordsLoadPlan
+from records_mover.records.types import RecordsHints
+from ...records.records_format import DelimitedRecordsFormat
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+PostgresCopyOptions = Dict[str, object]
+
+# https://www.postgresql.org/docs/9.2/sql-copy.html
+
+# https://www.postgresql.org/docs/9.3/multibyte.html
+postgres_encoding_names = {
+    'UTF8': 'UTF8',
+    # UTF-16 not supported
+    # https://www.postgresql.org/message-id/20051028061108.GB30916%40london087.server4you.de
+    'LATIN1': 'LATIN1',
+    'CP1252': 'WIN1252',
+}
+
+
+def needs_csv_format(hints: RecordsHints) -> bool:
+    # This format option is used for importing and exporting the Comma
+    # Separated Value (CSV) file format used by many other programs,
+    # such as spreadsheets. Instead of the escaping rules used by
+    # PostgreSQL's standard text format, it produces and recognizes
+    # the common CSV escaping mechanism.
+
+    # The values in each record are separated by the DELIMITER
+    # character. If the value contains the delimiter character, the
+    # QUOTE character, the NULL string, a carriage return, or line
+    # feed character, then the whole value is prefixed and suffixed by
+    # the QUOTE character, and any occurrence within the value of a
+    # QUOTE character or the ESCAPE character is preceded by the
+    # escape character. You can also use FORCE_QUOTE to force quotes
+    # when outputting non-NULL values in specific columns.
+    if hints['header-row'] or (hints['quoting'] is not None):
+        return True
+
+    return False
+
+
+def postgres_copy_options_common(unhandled_hints: Set[str],
+                                 hints: RecordsHints,
+                                 fail_if_cant_handle_hint: bool,
+                                 postgres_options: PostgresCopyOptions) -> None:
+
+    # TODO: Get this to work on read and write and test what we
+    # produce vs what we can accept
+    quiet_remove(unhandled_hints, 'datetimeformattz')
+
+    # TODO: Get this to work on read and write and test what we
+    # produce vs what we can accept
+    quiet_remove(unhandled_hints, 'datetimeformat')
+
+    # TODO: Get this to work on read and write and test what we
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker